homeassistant/components/iqvia/sensor.py

```diff
@@ -163,16 +224,22 @@ def update_from_latest_data(self) -> None:
             return
 
         try:
-            if self._type in (TYPE_ALLERGY_TODAY, TYPE_ALLERGY_TOMORROW):
+            if self.entity_description.key in (
+                TYPE_ALLERGY_TODAY,
+                TYPE_ALLERGY_TOMORROW,
+            ):
```

Comment:
It likely won't make much of a difference, but you could use an in-place `set`. In theory that would be slightly faster. Same for the `in` comparisons below.